File: CyberHunters/Assets/_SPECTRUM/scripts/doorPuzzle.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class doorPuzzle : MonoBehaviour {

    public GameObject doorGear;

    public GameObject parent;


    public void Use() 
    {
        foreach (Transform child in transform)
        {
            GameObject.Destroy(child.gameObject);
        }

        if(doorGear != null)
        {
            doorGear.SetActive(true);
        }
    }
      
        

    

}

File: Java/Assets/Scripts/doorPuzzle.java
import unityengine.*;

public class doorPuzzle extends MonoBehaviour
{

	public GameObject doorGear;

	public final void Use()
	{
		for (Transform child : transform)
		{
			GameObject.Destroy(child.gameObject);
		}

		if (doorGear != null)
		{
			doorGear.SetActive(true);
		}
	}





}